### ISSUE TITLE
Allow comments between fall-through case statement in a switch

### ISSUE THREAD

**Issue by dimaqq** [eslint/eslint#13829]
**Tell us about your environment**

* **ESLint Version:** v7.13.0
* **Node Version:** v14.9.0
* **npm Version:** 6.14.8

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
default parser;
following options:
```
    "parserOptions": {
      "ecmaFeatures": {
        "jsx": true
      },
      "ecmaVersion": 2020,
      "sourceType": "module"
    },
```

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
  "eslintConfig": {
    "env": {
      "browser": true,
      "es6": true,
      "jest": true
    },
    "extends": [
      "eslint:recommended",
      "plugin:react/recommended"
    ],
    "parserOptions": {
      "ecmaFeatures": {
        "jsx": true
      },
      "ecmaVersion": 2020,
      "sourceType": "module"
    },
    "plugins": [
      "react",
      "react-hooks"
    ],
    "rules": {
      "indent": [
        "warn",
        2
      ],
      "linebreak-style": [
        "error",
        "unix"
      ],
      "quotes": [
        "warn",
        "double",
        {
          "allowTemplateLiterals": true
        }
      ],
      "semi": [
        "error",
        "always"
      ],
      "comma-dangle": [
        "warn",
        "always-multiline"
      ],
      "react-hooks/rules-of-hooks": "error",
      "max-len": [
        "warn",
        120
      ],
      "no-mixed-operators": "off",
      "object-curly-spacing": [
        "warn",
        "never"
      ],
      "template-curly-spacing": [
        "warn",
        "always"
      ],
      "no-template-curly-in-string": "warn"
    },
    "settings": {
      "react": {
        "version": "detect"
      }
    }
  },
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

eslint doesn't like comments between case statements, like below:

```js
export default (x) => {
  switch (x) {
  case "one":
  // blah
  case "two":
  // blah
  default:
    console.log("barf");
  }
};
```

```bash
> yarn run eslint b.js
yarn run v1.22.4
$ /Users/uuu/proj/node_modules/.bin/eslint b.js

/Users/uuu/proj/b.js
  5:3  error  Expected a 'break' statement before 'case'     no-fallthrough
  7:3  error  Expected a 'break' statement before 'default'  no-fallthrough

✖ 2 problems (2 errors, 0 warnings)
```

**What did you expect to happen?**

Meanwhile, semantically equivalent code without comments is OK:
```
export default (x) => {
  switch (x) {
  case "one":
  case "two":
  default:
    console.log("barf");
  }
};
```

**Are you willing to submit a pull request to fix this bug?**
I'm not sure I know how...

**Comment by mdjermanovic** [comment]
Hi @dimaqq, thanks for the issue!

This is intended behavior, please see #5799.

Comments are allowed only on the same line as `case`:

```js
/* eslint no-fallthrough: ["error"] */

export default (x) => {
  switch (x) {
  case "one": // blah
  case "two": // blah
  default:
    console.log("barf");
  }
};
```

[lint-free demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWZhbGx0aHJvdWdoOiBbXCJlcnJvclwiXSAqL1xuXG5leHBvcnQgZGVmYXVsdCAoeCkgPT4ge1xuICBzd2l0Y2ggKHgpIHtcbiAgY2FzZSBcIm9uZVwiOiAvLyBibGFoXG4gIGNhc2UgXCJ0d29cIjogLy8gYmxhaFxuICBkZWZhdWx0OlxuICAgIGNvbnNvbGUubG9nKFwiYmFyZlwiKTtcbiAgfVxufTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

An alternative is to add a text that matches [commentPattern](https://eslint.org/docs/rules/no-fallthrough#commentpattern) to each comment.

**Comment by dimaqq** [comment]
Oh, bummer. I disagree with that decision, but I guess there's not much I can do.

My use case was relatively large, multi-line comments.
For the time being, I've just moved them at the top of the fall-through block, so I guess I can live with this.

**Comment by nzakas** [comment]
Working as intended, so closing.
